### PR TITLE
mgr/orchestrator: added useful attributes to ServiceDescription

### PR DESCRIPTION
**Pull request by rjfd**
Adds the `version`, `config_location`, and `service_url` to the description of a service.

These attributes are needed for services like NFS, or iSCSI, so that orchestrator clients can query the orchestrator about its services and then use the above information to manage the services.

Signed-off-by: Ricardo Dias <rdias@suse.com>

- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by LenzGr** [comment]
Small typo in the commit message: s/usefull/useful/

File: src/pybind/mgr/orchestrator.py [hunk 1]
```diff
@@ -333,6 +333,18 @@ def __init__(self):
         # The type of service (osd, mon, mgr, etc.)
         self.service_type = None
 
+        # Service version that was deployed
+        self.version = None
+
+        # Location of the configuration of the service, it might be
+        # a filesystem path, or a rados object path (pool/name).
```
Comment (sebastian-philipp) [inline src/pybind/mgr/orchestrator.py]: If it's a _filesystem path_, this is going to be a host dependent path with no meaning on other systems. I'd be :-1: for exposing filesystem paths of remote machines here. This should IMO be an implementation detail of the external orchestrator and not published to the cli.

Comment (rjfd) [reply src/pybind/mgr/orchestrator.py]: Fair enough. Then what if I rename the attribute to `rados_config_location` and it's just meant for rados objects?

Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator.py]: Sounds good!

Comment (tserong) [reply src/pybind/mgr/orchestrator.py]: This is a specific instance of a service, running on a particular host, so isn't it OK for a filesystem path to be host-dependent?

Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator.py]: Sure it is.

But why do we want to expose them? No one, except for the external orchestrator, should directly touch that file. No user should ssh into a running container and modify that file.

Comment (rjfd) [reply src/pybind/mgr/orchestrator.py]: > No user should ssh into a running container and modify that file.

The orchestrator is not meant just for container environments, right?
Having the location of where the orchestrator implementation stored the configuration of a particular service instance might be useful for sysadmins to debug problems.

FYI, in my current use case, I only store rados objects paths in this attribute.

Comment (tserong) [reply src/pybind/mgr/orchestrator.py]: Fair enough. That raises the question though, if external orchestrators are the only things meant to touch service configuration, why expose configuration that's stored in rados objects to orchestrator clients? Doesn't that blur the lines between what's in control of service configuration (external orchestrator vs. orchestrator client)?

Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator.py]: For rados objects paths, we have a real use case: the dashboard can directly set the reados objects without going through the external ouchestrator.

Comment (rjfd) [reply src/pybind/mgr/orchestrator.py]: What I understood from all the discussions about what are the responsibilities of the orchestrator was that the orchestrator is only responsible for deployment/undeployment. The management of running services was done by external tools, such as the dashboard.

That is why I added this attributes to the `ServiceDescription` class, to allow external tools to take over after the orchestrator deploys the services.

Comment (tserong) [reply src/pybind/mgr/orchestrator.py]: > What I understood from all the discussions about what are the responsibilities of the orchestrator was that the orchestrator is only responsible for deployment/undeployment.

Depends on the orchestrator and the service, e.g. DeepSea knows how to create RGW users, which is arguably management of a running service, but that you would of course also want to do through the dashboard.

Anyway, it's all good, I understand the rationale now, so fine by me :-)

**Comment by rjfd** [comment]
@sebastian-philipp @LenzGr addressed your comments and pushed changes

File: src/pybind/mgr/orchestrator.py [hunk 2]
```diff
@@ -333,6 +333,17 @@ def __init__(self):
         # The type of service (osd, mon, mgr, etc.)
         self.service_type = None
 
+        # Service version that was deployed
+        self.version = None
+
+        # Location of the service configuration when stored in rados
+        # object. Format: "<pool_name>/<object_name>"
+        self.rados_config_location = None
```
Comment (jtlayton) [inline src/pybind/mgr/orchestrator.py]: What about namespaces here? FWIW, ganesha has crafted its own rados:// URL type for use internally:

```rados://<pool>/[<namespace/>]<object>```

A 2 component URL there means that the namespace is the default one. Maybe this should be displayed in a similar format?

Comment (rjfd) [reply src/pybind/mgr/orchestrator.py]: I agree, will change the comment accordingly. Should we also keep the `rados://' part?

Comment (sebastian-philipp) [reply src/pybind/mgr/orchestrator.py]: I'd be :+1: for using a full url here

**Comment by sebastian-philipp** [comment]
Looks innocent to me. As long as we set this to a sane value in the rook module.

**Comment by rjfd** [comment]
@sebastian-philipp added service status attributes in commit 92776e0

**Pull request merged**